cmd/metrics.go

```diff
@@ -94,6 +106,40 @@ func (c *minioCollector) Collect(ch chan<- prometheus.Metric) {
 	healingMetricsPrometheus(ch)
 }
 
+func nodeHealthMetricsPrometheus(ch chan<- prometheus.Metric) {
+	nodesUp, nodesDown := 1, 1
+	// This runs a basic health check across nodes.
+	// The actual io path for peer communication has its own logic
+	// for marking peers down. This does not piggy back on that logic
+	// this is to allow for reporting of node online status in the absence
+	// of traffic.
+	for _, c := range globalNotificationSys.peerClients {
+		if c.restClient.HealthCheckFn != nil {
+			if c.restClient.HealthCheckFn() {
+				nodesUp++
+				continue
+			}
+			nodesDown++
+		}
+	}
```

Comment:
This is not needed - there is already code for this which can be re-used written at cmd/admin-handlers.go

```
        mode := "online"
        server := getLocalServerProperty(globalEndpoints, r)
        servers := globalNotificationSys.ServerInfo()
        servers = append(servers, server)
```

Reply:
Will do. This check is slightly different in the error path from the one used by the client as all errors are treated as down here.